admin.go

```diff
@@ -3,14 +3,42 @@ package intercom
 import (
 	"encoding/json"
 	"fmt"
+	"reflect"
+	"strconv"
 )
 
 // Admin represents an Admin in Intercom.
 type Admin struct {
-	ID    json.Number `json:"id"`
-	Type  string      `json:"type"`
-	Name  string      `json:"name"`
-	Email string      `json:"email"`
+	ID    string `json:"id"`
+	Type  string `json:"type"`
+	Name  string `json:"name"`
+	Email string `json:"email"`
+}
+
+// UnmarshalJSON custom SocialProfile unmarshaller
+func (a *Admin) UnmarshalJSON(data []byte) error {
+	s := struct {
+		ID    interface{} `json:"id"`
+		Type  string      `json:"type"`
+		Name  string      `json:"name"`
+		Email string      `json:"email"`
+	}{}
+	if err := json.Unmarshal(data, &s); err != nil {
+		return err
+	}
+	a.Type = s.Type
+	a.Name = s.Name
+	a.Email = s.Email
+
+	// only 3 possible combinations nil, string, float64
+	if s.ID != nil {
+		if reflect.TypeOf(s.ID).Kind() == reflect.Float64 {
```

Comment:
would a type switch work here instead of reflect?

Reply:
yes, but as far as I know it's the same difference...happy to change if you want

Reply:
aah, using reflect when avoidable is just bad karma, you know

Reply:
done